macros/adapters/default/page_views/snowplow_page_views.sql

```diff
@@ -329,6 +329,7 @@ prep as (
       and (
         a.useragent not {{ snowplow.similar_to('%(bot|crawl|slurp|spider|archiv|spinn|sniff|seo|audit|survey|pingdom|worm|capture|(browser|screen)shots|analyz|index|thumb|check|facebook|PingdomBot|PhantomJS|YandexBot|Twitterbot|a_archiver|facebookexternalhit|Bingbot|BingPreview|Googlebot|Baiduspider|360(Spider|User-agent)|semalt)%') }}
         or a.useragent is null
+        or a.br_type not {{ snowplow.similar_to('%(bot|crawl|slurp|spider|archiv|spinn|sniff|seo|audit|survey|pingdom|worm|capture|(browser|screen)shots|analyz|index|thumb|check|facebook|PingdomBot|PhantomJS|YandexBot|Twitterbot|a_archiver|facebookexternalhit|Bingbot|BingPreview|Googlebot|Baiduspider|360(Spider|User-agent)|semalt)%') }}
```

Comment:
I think `br_type` is provided by the (now deprecated) [user agent utils enrichment](https://github.com/snowplow/snowplow/wiki/user-agent-utils-enrichment). This is superseded by the [ua parser enrichement](https://github.com/snowplow/snowplow/wiki/ua-parser-enrichment).

I did some digging here (this has been on my list for a while) and I have a couple of thoughts:
1. `br_type` can definitely be null (eg. if the enrichment is disabled). Can you add an additional check for `or a.br_type` is null? We want to include records with a null `br_type`
2. I think we can find an enumeration of the possible values for this field [in the user-agent-utils library](https://github.com/HaraldWalker/user-agent-utils/blob/trunk/src/main/java/eu/bitwalker/useragentutils/BrowserType.java#L50-L75)

A quick query on Fishtown's Snowplow data revealed these distinct values for `br_type` in the `events` table:
```
unknown
Browser
Browser (mobile)
Robot
```

So, putting this all together, I think we can do something like:
```
or coalesce(a.br_type, 'unknown') != 'Robot'
```

Do you buy all of that?

Reply:
err, we might want to make this a separate predicate event:

```
where (a.br_family != 'Robot/Spider' or a.br_family is null)
and (
  useragent filters
)
and (
  br_type logic
)
```

Reply:
😬  Sorry for being a bit overzealous in trying to address https://github.com/fishtown-analytics/snowplow/issues/25

I'll get this updated today 👍

Reply:
updated in the latest version to filter out bot/crawler and robot explicitly. This format could easily be updated to add another _thing_ to the list down the road. LMK if that doesn't work.

Reply:
I think this line will filter out records where `br_type` is null:
```
and a.br_type not in ('Bot/Crawler', 'Robot')
```

Can you instead change this line to coalesce `a.br_type` to some non-null value (like `'unknown'`) to let `null` br_type values pass through? This can happen if this particular enrichment is disabled in a Snowplow pipeline.

with sample as (
	select 1 as id, 'Robot' as br_type union all
	select 2 as id, 'Bot/Crawler' as br_type union all
	select 3 as id, 'Browser' as br_type union all
	select 4 as id, null as br_type
)
select *
from sample
where br_type not in ('Robot', 'Bot/Crawler')
--where coalesce(br_type, 'unknown') not in ('Robot', 'Bot/Crawler')

Reply:
Sorry for misunderstanding! It should be fixed in the latest version!